### DISCUSSION TITLE
Can't decode raw JWT on jwt.io

### DISCUSSION THREAD

**Discussion by rizchaerul** [nextauthjs/next-auth#4328] [Help]
So I create this `/api/jwt` endpoint for getting raw JWT
```
export default async function handler(req: any, res: any) {
    const rawToken = await getToken({ req, raw: true });
    const decodedToken = await getToken({ req, raw: false });

    console.log("JSON Web Token", decodedToken);
    res.send({
        rawToken,
        decodedToken,
    });
}
```

And the result is this
```
{
  "rawToken": "eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..B2YBggHYYZOkDkNy.1goDsgCvMJwRu_bCGtKFWU99vKac1brHpHdI9nfUnKTxFQofTwa14y-O3oeo8CSovQpa877RTinzlJ9X0Cc4CfRHimaT_JtSbVeDpXukjA23ZlJmAzkadlBbEMsSPbTsRE9uc0NbtxCdFOBZulndTUebaWO9a0_TA0309RHpb83fs8Ce9l2ml3rAGkiSrZXgfI8SLqsNtrNmqQKkFsSzH6p8fBU.GINILOI3JkUeF7QJLm6zEw",
  "decodedToken": {
  "sub": "892ad5b9-e90e-46cf-9dba-575448a16033",
  "name": "User",
  "email": "user@email.com",
  "iat": 1649123526,
  "exp": 1651715526,
  "jti": "d4b004b6-8298-40a3-a543-5c0f2062de91"
  }
}
```

When I paste the rawToken to jwt.io, this shows up
![image](https://user-images.githubusercontent.com/72289631/161664244-dc77704e-0e88-41b8-a595-4b2dc51784f1.png)

What am I doing wrong?

**Comment by productdevbook** [comment]
When JSON Web Tokens are enabled, they are encrypted by default (JWE) with A256GCM

You need to decode the incoming raw because it is encrypted.

https://github.com/nextauthjs/next-auth/blob/a72f1b6d21da89b4223542006c865862635c027b/packages/next-auth/src/jwt/index.ts#L67

jwt.io In short, it cannot solve it.
